ui/src/map/layers/operational-point/operational-points-area-placing-layer.ts

```diff
@@ -0,0 +1,101 @@
+import { Feature } from 'ol';
+import { Polygon as OlPolygon } from 'ol/geom';
+import Style from 'ol/style/Style';
+import CircleStyle from 'ol/style/Circle.js';
+import Fill from 'ol/style/Fill';
+import { LayerItemSearchResult, MapLayer, SearchItemsOptions } from 'map/layers/utils/layer-model';
+import { coordsToPolygon, Polygon, Rectangle } from 'model/geometry';
+import {
+    findMatchingOperationalPoints,
+    operationalPointAreaPolygonStyle,
+} from 'map/layers/operational-point/operational-points-layer-utils';
+import {
+    createLayer,
+    GeoviiteMapLayer,
+    getFeatureCoords,
+    loadLayerData,
+    pointToCoords,
+} from 'map/layers/utils/layer-utils';
+import { PlacingOperationalPointArea } from 'linking/linking-model';
+import { getOperationalPoint } from 'track-layout/layout-operational-point-api';
+import { LayoutContext } from 'common/common-model';
+import { Modify } from 'ol/interaction';
+import { doubleClick } from 'ol/events/condition';
+import OlMap from 'ol/Map';
+
+const LAYER_NAME = 'operational-points-area-placing-layer';
+
+export const createOperationalPointsAreaPlacingLayer = (
+    existingOlLayer: GeoviiteMapLayer<OlPolygon>,
+    linkingState: PlacingOperationalPointArea | undefined,
+    layoutContext: LayoutContext,
+    map: OlMap,
+    modifyInteraction: Modify | undefined,
+    setModifyInteraction: React.Dispatch<React.SetStateAction<Modify | undefined>>,
+    onSetOperationalPointPolygon: (polygon: Polygon) => void,
+    onLoadingData: (loading: boolean) => void,
+): MapLayer => {
+    const { layer, source, isLatest } = createLayer(LAYER_NAME, existingOlLayer, true);
+    const selectedOperationalPointId = linkingState?.operationalPoint?.id;
+
+    setModifyInteraction((previousModify) => {
+        if (previousModify) {
+            map.removeInteraction(previousModify);
+        }
+        const modify = new Modify({
+            source: source,
+            deleteCondition: doubleClick,
+            style: new Style({
+                image: new CircleStyle({
+                    radius: 5,
+                    fill: new Fill({
+                        color: '#009BFF',
+                    }),
+                }),
+            }),
+        });
+        modify.on('modifyend', (event) => {
```

Comment:
Tämä siis ajautuu aina kun hiiren ykkösnapista päästää irti. Tallennetaan tällöin polygonin senhetkinen tila reduxiin